src/app/loans/loans-view/loan-tranche-details/loan-tranche-details.component.ts

```diff
@@ -10,26 +19,41 @@ export class LoanTrancheDetailsComponent implements OnInit {
   loanDetails: any;
   return: any;
   status: any;
-  totalDisbursedAmount: any;
   count: number;
   expectedDisbursementColumns: string[] = ['expected disbursement on', 'disbursed on', 'principal', 'action'];
   emivariationColumns: string[] = ['emi amount variation from', 'fixed emi amount'];
 
+  loanId: number;
+  currentPrincipalAmount: number;
+  minDate = new Date(2000, 0, 1);
+  maxDate = new Date(2100, 0, 1);
+  disbursementDataSource: {}[] = [];
+  totalMultiDisbursed: number = null;
+  pristine = true;
+
   /**
    * Retrieves the loans data from `resolve`.
    * @param {ActivatedRoute} route Activated Route.
    */
-  constructor(private route: ActivatedRoute) {
+  constructor(private route: ActivatedRoute,
+    public dialog: MatDialog,
+    private loanServices: LoansService,
+    private settingsService: SettingsService,
+    private dateUtils: Dates) {
     this.route.parent.data.subscribe((data: { loanDetailsData: any }) => {
+      this.loanId = data.loanDetailsData.id;
       this.loanDetails = data.loanDetailsData;
+      this.disbursementDataSource = data.loanDetailsData.disbursementDetails;
+      this.currentPrincipalAmount = this.loanDetails.approvedPrincipal;
     });
   }
 
   ngOnInit() {
+    this.maxDate = this.settingsService.maxFutureDate;
     this.status = this.loanDetails.status.value;
   }
 
-  showAddDeleteTrancheButtons(action: string) {
+  showAddTrancheButtons() {
```

Comment:
Please disable if the "expected disbursement details" are disabled explicitly on the loan product